perun/logic/index.py

```diff
@@ -655,6 +657,42 @@ def get_profile_number_for_minor(base_dir, minor_version):
     else:
         return {'all': 0}
 
+
+def load_custom_index(index_path):
+    """Loads the content of a custom index file (e.g. temp or stats) as a dictionary.
+
+    The index is json-formatted and compressed. In case the index cannot be read for some reason,
+    an empty dictionary is returned.
+
+    :param str index_path: path to the index file
+
+    :return: the decompressed and json-decoded index content.
+    """
+    # Create and init the index file if it does not exist yet
+    if not os.path.exists(index_path):
+        store.touch_file(index_path)
+        save_custom_index(index_path, {})
+    # Open and load the file
+    try:
+        with open(index_path, 'rb') as index_handle:
+            return json.loads(store.read_and_deflate_chunk(index_handle))
```

Comment:
Do you mean the perun.utils.streams.py? I double checked but there is no such function in the module.

Reply:
Hm, I remeber that I used to have it somewhere, but now I see that it was moved probably because of some dependencies or something. I may refactor it in my next release though, so don't worry about this comment.